### ISSUE TITLE
Fix `declaration-property-value-no-unknown` parse error for multiline interpolation

### ISSUE THREAD

**Issue by FloEdelmann** [stylelint/stylelint#7403]
### What minimal example or steps are needed to reproduce the bug?

```scss
.foo {
  color: #{
    some-scss-function()
  };
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-property-value-no-unknown": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint file.scss
```

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "^15.11.0",
  "stylelint-config-standard-scss": "^11.1.0"
}
```

### What did you expect to happen?

No error, since interpolations should be ignored in the rule:

https://github.com/stylelint/stylelint/blob/da7ce21272d34928f40bbc68c1e01be1c076bb8a/lib/rules/declaration-property-value-no-unknown/index.mjs#L122

### What actually happened?

A problem was reported:

```
Cannot parse property value "#{
    some-scss-function()
  }" for property "color" (declaration-property-value-no-unknown)
```

### Do you have a proposal to fix the bug?

I think the regex here needs to be changed to also take newline characters into account:

https://github.com/stylelint/stylelint/blob/da7ce21272d34928f40bbc68c1e01be1c076bb8a/lib/utils/hasScssInterpolation.mjs#L1

**Comment by jeddy3** [comment]
@FloEdelmann Thanks for the report and for using the template.

The rule is [only for CSS](https://stylelint.io/user-guide/rules/declaration-property-value-no-unknown/):

> This rule is only appropriate for CSS. You should not turn it on for CSS-like languages, such as Sass or Less, as they have their own syntaxes.

Having said that, there's no harm in improving the interpolation to migrate this particular issue.  You may encounter other issue trying to use the rule for SCSS, though.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
